src/lib/fragments/Pileup.react.js

```diff
@@ -0,0 +1,89 @@
+import React, {Component} from 'react';
+import {propTypes, defaultProps} from '../components/Pileup.react';
+import pileup from 'pileup';
+
+import '../../../node_modules/pileup/style/pileup.css';
+
+/**
+ * The Pileup component is an genome visualization component
+ * developed by the the Hammerlab. It uses an
+ * example integration of pileup.js and React (https://www.npmjs.com/package/pileup).
+ */
+export default class Pileup extends Component {
+    constructor(props) {
+        super(props);
+        this.ref = React.createRef();
+        this.pileup = null;
+    }
+
+    parseTracks(reference, tracks) {
+        var referenceTrack = {
+            viz: pileup.viz.genome(),
+            isReference: true,
+            data: pileup.formats.twoBit({
+                url: reference.url,
+            }),
+            name: reference.label,
+        };
+
+        // make list of pileup sources
+        var sources = [referenceTrack];
+
+        // add in optional tracks
+        for (var i = 0; i < tracks.length; i++) {
+            var track = tracks[i];
+
+            var newTrack = {
+                viz: pileup.viz[track.viz](),
+                isReference: false,
+                data: null,
+                name: track.label,
+            };
+
+            // Make sure source exists and it is a valid pileup format
+            // Source may not exist for scale or location tracks
+            if (('source' in track) & (pileup.formats[track.source] !== null)) {
+                newTrack.data = pileup.formats[track.source](
+                    track.sourceOptions
+                );
+            }
+            sources.push(newTrack);
+        }
+        return sources;
+    }
+
+    createPileupBrowser() {
+        // var pileupContainer = this.ref.current;
+        var pileupOptions = {
+            range: this.props.range,
+            tracks: this.parseTracks(this.props.reference, this.props.tracks),
+        };
+        if (this.pileup !== null) {
```

Comment:
It might make more sense to use `isNil` from the `ramda` library here, which would make sure that `this.pileup` is not `null` _or_ `undefined` before executing the codeblock for a little more consistency.

```
import { isNil } from 'ramda';
```